File: releases/v3.3.0/schemaorggo/flight.go
package schemaorggo

import "encoding/json"

// Flight see : https://schema.org/Flight
type Flight struct {

	// With properties from Intangible see : https://schema.org/Intangible
	//

	// With properties from Thing see : https://schema.org/Thing
	//

	// AdditionalType see : https://schema.org/additionalType
	// An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	// types : URL
	AdditionalType []string `json:"additionalType,omitempty"`

	// Aircraft see : https://schema.org/aircraft
	// The kind of aircraft (e.g., &quot;Boeing 747&quot;).
	// types : Text Vehicle
	Aircraft []interface{} `json:"aircraft,omitempty"`

	// AlternateName see : https://schema.org/alternateName
	// An alias for the item.
	// types : Text
	AlternateName []string `json:"alternateName,omitempty"`

	// ArrivalAirport see : https://schema.org/arrivalAirport
	// The airport where the flight terminates.
	// types : Airport
	ArrivalAirport []*Airport `json:"arrivalAirport,omitempty"`

	// ArrivalGate see : https://schema.org/arrivalGate
	// Identifier of the flight&#39;s arrival gate.
	// types : Text
	ArrivalGate []string `json:"arrivalGate,omitempty"`

	// ArrivalTerminal see : https://schema.org/arrivalTerminal
	// Identifier of the flight&#39;s arrival terminal.
	// types : Text
	ArrivalTerminal []string `json:"arrivalTerminal,omitempty"`

	// ArrivalTime see : https://schema.org/arrivalTime
	// The expected arrival time.
	// types : DateTime
	ArrivalTime []DateTime `json:"arrivalTime,omitempty"`

	// BoardingPolicy see : https://schema.org/boardingPolicy
	// The type of boarding policy used by the airline (e.g. zone-based or group-based).
	// types : BoardingPolicyType
	BoardingPolicy []*BoardingPolicyType `json:"boardingPolicy,omitempty"`

	// Carrier see : https://schema.org/carrier
	// &#39;carrier&#39; is an out-dated term indicating the &#39;provider&#39; for parcel delivery and flights.
	// types : Organization
	Carrier []*Organization `json:"carrier,omitempty"`

	// DepartureAirport see : https://schema.org/departureAirport
	// The airport where the flight originates.
	// types : Airport
	DepartureAirport []*Airport `json:"departureAirport,omitempty"`

	// DepartureGate see : https://schema.org/departureGate
	// Identifier of the flight&#39;s departure gate.
	// types : Text
	DepartureGate []string `json:"departureGate,omitempty"`

	// DepartureTerminal see : https://schema.org/departureTerminal
	// Identifier of the flight&#39;s departure terminal.
	// types : Text
	DepartureTerminal []string `json:"departureTerminal,omitempty"`

	// DepartureTime see : https://schema.org/departureTime
	// The expected departure time.
	// types : DateTime
	DepartureTime []DateTime `json:"departureTime,omitempty"`

	// Description see : https://schema.org/description
	// A description of the item.
	// types : Text
	Description []string `json:"description,omitempty"`

	// DisambiguatingDescription see : https://schema.org/disambiguatingDescription
	// A sub property of description. A short description of the item used to disambiguate from other, similar items. Information from other properties (in particular, name) may be necessary for the description to be useful for disambiguation.
	// types : Text
	DisambiguatingDescription []string `json:"disambiguatingDescription,omitempty"`

	// EstimatedFlightDuration see : https://schema.org/estimatedFlightDuration
	// The estimated time the flight will take.
	// types : Text Duration
	EstimatedFlightDuration []interface{} `json:"estimatedFlightDuration,omitempty"`

	// FlightDistance see : https://schema.org/flightDistance
	// The distance of the flight.
	// types : Text Distance
	FlightDistance []interface{} `json:"flightDistance,omitempty"`

	// FlightNumber see : https://schema.org/flightNumber
	// The unique identifier for a flight including the airline IATA code. For example, if describing United flight 110, where the IATA code for United is &#39;UA&#39;, the flightNumber is &#39;UA110&#39;.
	// types : Text
	FlightNumber []string `json:"flightNumber,omitempty"`

	// Identifier see : https://schema.org/identifier
	// The identifier property represents any kind of identifier for any kind of [[Thing]], such as ISBNs, GTIN codes, UUIDs etc. Schema.org provides dedicated properties for representing many of these, either as textual strings or as URL (URI) links. See [background notes](/docs/datamodel.html#identifierBg) for more details.
	//
	// types : URL Text PropertyValue
	Identifier []interface{} `json:"identifier,omitempty"`

	// Image see : https://schema.org/image
	// An image of the item. This can be a [[URL]] or a fully described [[ImageObject]].
	// types : URL ImageObject
	Image []interface{} `json:"image,omitempty"`

	// MainEntityOfPage see : https://schema.org/mainEntityOfPage
	// Indicates a page (or other CreativeWork) for which this thing is the main entity being described. See [background notes](/docs/datamodel.html#mainEntityBackground) for details.
	// types : CreativeWork URL
	MainEntityOfPage []interface{} `json:"mainEntityOfPage,omitempty"`

	// MealService see : https://schema.org/mealService
	// Description of the meals that will be provided or available for purchase.
	// types : Text
	MealService []string `json:"mealService,omitempty"`

	// Name see : https://schema.org/name
	// The name of the item.
	// types : Text
	Name []string `json:"name,omitempty"`

	// PotentialAction see : https://schema.org/potentialAction
	// Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	// types : Action
	PotentialAction []*Action `json:"potentialAction,omitempty"`

	// Provider see : https://schema.org/provider
	// The service provider, service operator, or service performer; the goods producer. Another party (a seller) may offer those services or goods on behalf of the provider. A provider may also serve as the seller.
	// types : Person Organization
	Provider []interface{} `json:"provider,omitempty"`

	// SameAs see : https://schema.org/sameAs
	// URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Wikidata entry, or official website.
	// types : URL
	SameAs []string `json:"sameAs,omitempty"`

	// Seller see : https://schema.org/seller
	// An entity which offers (sells / leases / lends / loans) the services / goods.  A seller may also be a provider.
	// types : Organization Person
	Seller []interface{} `json:"seller,omitempty"`

	// Url see : https://schema.org/url
	// URL of the item.
	// types : URL
	Url []string `json:"url,omitempty"`

	// WebCheckinTime see : https://schema.org/webCheckinTime
	// The time when a passenger can check into the flight online.
	// types : DateTime
	WebCheckinTime []DateTime `json:"webCheckinTime,omitempty"`
}

func (v Flight) MarshalJSON() ([]byte, error) {
	type Alias Flight

	b, err := json.Marshal((Alias)(v))
	if err != nil {
		return nil, err
	}

	return append([]byte("{\"@context\":\"http://schema.org\",\"@type\":\"Flight\","), b[1:]...), nil
}

File: releases/v2.2.0/schemaorgphp/classes/flight.php
<?php
namespace SchemaOrg;

// Flight see : https://schema.org/Flight
class Flight implements \JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type    = 'Flight';

	/**
	 * With properties from Intangible see : https://schema.org/Intangible
	 */

	/**
	 * With properties from Thing see : https://schema.org/Thing
	 */


	/**
	 * An additional type for the item, typically used for adding more specific types from external vocabularies in microdata syntax. This is a relationship between something and a class that the thing is in. In RDFa syntax, it is better to use the native RDFa syntax - the &#39;typeof&#39; attribute - for multiple types. Schema.org tools may have only weaker understanding of extra types, in particular those defined externally.
	 * see : https://schema.org/additionalType
	 *
	 * @var string | string[]
	 */
	public $additional_type;

	/**
	 * The kind of aircraft (e.g., &quot;Boeing 747&quot;).
	 * see : https://schema.org/aircraft
	 *
	 * @var string | string[] | \Vehicle | \Vehicle[]
	 */
	public $aircraft;

	/**
	 * An alias for the item.
	 * see : https://schema.org/alternateName
	 *
	 * @var string | string[]
	 */
	public $alternate_name;

	/**
	 * The airport where the flight terminates.
	 * see : https://schema.org/arrivalAirport
	 *
	 * @var \Airport | \Airport[]
	 */
	public $arrival_airport;

	/**
	 * Identifier of the flight&#39;s arrival gate.
	 * see : https://schema.org/arrivalGate
	 *
	 * @var string | string[]
	 */
	public $arrival_gate;

	/**
	 * Identifier of the flight&#39;s arrival terminal.
	 * see : https://schema.org/arrivalTerminal
	 *
	 * @var string | string[]
	 */
	public $arrival_terminal;

	/**
	 * The expected arrival time.
	 * see : https://schema.org/arrivalTime
	 *
	 * @var string | string[]
	 */
	public $arrival_time;

	/**
	 * The type of boarding policy used by the airline (e.g. zone-based or group-based).
	 * see : https://schema.org/boardingPolicy
	 *
	 * @var \BoardingPolicyType | \BoardingPolicyType[]
	 */
	public $boarding_policy;

	/**
	 * &#39;carrier&#39; is an out-dated term indicating the &#39;provider&#39; for parcel delivery and flights.
	 * see : https://schema.org/carrier
	 *
	 * @var \Organization | \Organization[]
	 */
	public $carrier;

	/**
	 * The airport where the flight originates.
	 * see : https://schema.org/departureAirport
	 *
	 * @var \Airport | \Airport[]
	 */
	public $departure_airport;

	/**
	 * Identifier of the flight&#39;s departure gate.
	 * see : https://schema.org/departureGate
	 *
	 * @var string | string[]
	 */
	public $departure_gate;

	/**
	 * Identifier of the flight&#39;s departure terminal.
	 * see : https://schema.org/departureTerminal
	 *
	 * @var string | string[]
	 */
	public $departure_terminal;

	/**
	 * The expected departure time.
	 * see : https://schema.org/departureTime
	 *
	 * @var string | string[]
	 */
	public $departure_time;

	/**
	 * A short description of the item.
	 * see : https://schema.org/description
	 *
	 * @var string | string[]
	 */
	public $description;

	/**
	 * The estimated time the flight will take.
	 * see : https://schema.org/estimatedFlightDuration
	 *
	 * @var string | string[] | \Duration | \Duration[]
	 */
	public $estimated_flight_duration;

	/**
	 * The distance of the flight.
	 * see : https://schema.org/flightDistance
	 *
	 * @var string | string[] | \Distance | \Distance[]
	 */
	public $flight_distance;

	/**
	 * The unique identifier for a flight including the airline IATA code. For example, if describing United flight 110, where the IATA code for United is &#39;UA&#39;, the flightNumber is &#39;UA110&#39;.
	 * see : https://schema.org/flightNumber
	 *
	 * @var string | string[]
	 */
	public $flight_number;

	/**
	 * An image of the item. This can be a &lt;a href=&quot;http://schema.org/URL&quot;&gt;URL&lt;/a&gt; or a fully described &lt;a href=&quot;http://schema.org/ImageObject&quot;&gt;ImageObject&lt;/a&gt;.
	 * see : https://schema.org/image
	 *
	 * @var string | string[] | \ImageObject | \ImageObject[]
	 */
	public $image;

	/**
	 * Indicates a page (or other CreativeWork) for which this thing is the main entity being described.
	 *       &lt;br /&gt;&lt;br /&gt;
	 *       See &lt;a href=&quot;/docs/datamodel.html#mainEntityBackground&quot;&gt;background notes&lt;/a&gt; for details.
	 *
	 * see : https://schema.org/mainEntityOfPage
	 *
	 * @var \CreativeWork | \CreativeWork[] | string | string[]
	 */
	public $main_entity_of_page;

	/**
	 * Description of the meals that will be provided or available for purchase.
	 * see : https://schema.org/mealService
	 *
	 * @var string | string[]
	 */
	public $meal_service;

	/**
	 * The name of the item.
	 * see : https://schema.org/name
	 *
	 * @var string | string[]
	 */
	public $name;

	/**
	 * Indicates a potential Action, which describes an idealized action in which this thing would play an &#39;object&#39; role.
	 * see : https://schema.org/potentialAction
	 *
	 * @var \Action | \Action[]
	 */
	public $potential_action;

	/**
	 * The service provider, service operator, or service performer; the goods producer. Another party (a seller) may offer those services or goods on behalf of the provider. A provider may also serve as the seller.
	 * see : https://schema.org/provider
	 *
	 * @var \Person | \Person[] | \Organization | \Organization[]
	 */
	public $provider;

	/**
	 * URL of a reference Web page that unambiguously indicates the item&#39;s identity. E.g. the URL of the item&#39;s Wikipedia page, Freebase page, or official website.
	 * see : https://schema.org/sameAs
	 *
	 * @var string | string[]
	 */
	public $same_as;

	/**
	 * An entity which offers (sells / leases / lends / loans) the services / goods.  A seller may also be a provider.
	 * see : https://schema.org/seller
	 *
	 * @var \Organization | \Organization[] | \Person | \Person[]
	 */
	public $seller;

	/**
	 * URL of the item.
	 * see : https://schema.org/url
	 *
	 * @var string | string[]
	 */
	public $url;

	/**
	 * The time when a passenger can check into the flight online.
	 * see : https://schema.org/webCheckinTime
	 *
	 * @var string | string[]
	 */
	public $web_checkin_time;

	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type'    => 'Flight',
		);

		$serialized = \SchemaOrg\json_serialize( $this->additional_type );
		if ( ! empty( $serialized ) ) {
			$out['additionalType'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->aircraft );
		if ( ! empty( $serialized ) ) {
			$out['aircraft'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->alternate_name );
		if ( ! empty( $serialized ) ) {
			$out['alternateName'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->arrival_airport );
		if ( ! empty( $serialized ) ) {
			$out['arrivalAirport'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->arrival_gate );
		if ( ! empty( $serialized ) ) {
			$out['arrivalGate'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->arrival_terminal );
		if ( ! empty( $serialized ) ) {
			$out['arrivalTerminal'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->arrival_time );
		if ( ! empty( $serialized ) ) {
			$out['arrivalTime'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->boarding_policy );
		if ( ! empty( $serialized ) ) {
			$out['boardingPolicy'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->carrier );
		if ( ! empty( $serialized ) ) {
			$out['carrier'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->departure_airport );
		if ( ! empty( $serialized ) ) {
			$out['departureAirport'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->departure_gate );
		if ( ! empty( $serialized ) ) {
			$out['departureGate'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->departure_terminal );
		if ( ! empty( $serialized ) ) {
			$out['departureTerminal'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->departure_time );
		if ( ! empty( $serialized ) ) {
			$out['departureTime'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->description );
		if ( ! empty( $serialized ) ) {
			$out['description'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->estimated_flight_duration );
		if ( ! empty( $serialized ) ) {
			$out['estimatedFlightDuration'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->flight_distance );
		if ( ! empty( $serialized ) ) {
			$out['flightDistance'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->flight_number );
		if ( ! empty( $serialized ) ) {
			$out['flightNumber'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->image );
		if ( ! empty( $serialized ) ) {
			$out['image'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->main_entity_of_page );
		if ( ! empty( $serialized ) ) {
			$out['mainEntityOfPage'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->meal_service );
		if ( ! empty( $serialized ) ) {
			$out['mealService'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->name );
		if ( ! empty( $serialized ) ) {
			$out['name'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->potential_action );
		if ( ! empty( $serialized ) ) {
			$out['potentialAction'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->provider );
		if ( ! empty( $serialized ) ) {
			$out['provider'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->same_as );
		if ( ! empty( $serialized ) ) {
			$out['sameAs'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->seller );
		if ( ! empty( $serialized ) ) {
			$out['seller'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->url );
		if ( ! empty( $serialized ) ) {
			$out['url'] = $serialized;
		}

		$serialized = \SchemaOrg\json_serialize( $this->web_checkin_time );
		if ( ! empty( $serialized ) ) {
			$out['webCheckinTime'] = $serialized;
		}

		return $out;
	}
}
